src/game/weapons.cpp

```diff
@@ -21,22 +23,21 @@ namespace weapons
     vector<int> weaplist;
     void buildweaplist(const char *str)
     {
-        vector<char *> list;
+        std::vector<std::string> list;
         explodelist(str, list);
         weaplist.shrink(0);
         loopv(list)
         {
             int weap = -1;
-            if(isnumeric(list[i][0])) weap = atoi(list[i]);
-            else loopj(W_ALL) if(!strcasecmp(weaptype[j].name, list[i]))
+            if(isnumeric(list[i][0])) weap = atoi(list[i].data());
```

Comment:
Would appreciate if you could clean up this "stuff everything into one line" stuff.